src/fondant/pipeline/compiler.py

```diff
@@ -353,48 +354,55 @@ def from_fondant_component_spec(
             },
         }
 
-        cleaned_component_name = fondant_component.sanitized_component_name
-
+        kfp_safe_name = (
+            re.sub(
+                "-+",
+                "-",
+                re.sub("[^-0-9a-z]+", "-", fondant_component.name.lower()),
+            )
+            .lstrip("-")
+            .rstrip("-")
+        )
         specification = {
             "components": {
                 "comp-"
-                + cleaned_component_name: {
-                    "executorLabel": "exec-" + cleaned_component_name,
+                + kfp_safe_name: {
+                    "executorLabel": "exec-" + kfp_safe_name,
                     "inputDefinitions": input_definitions,
                 },
             },
             "deploymentSpec": {
                 "executors": {
                     "exec-"
-                    + cleaned_component_name: {
+                    + kfp_safe_name: {
                         "container": {
                             "command": command,
                             "image": image_uri,
                         },
                     },
                 },
             },
-            "pipelineInfo": {"name": cleaned_component_name},
+            "pipelineInfo": {"name": kfp_safe_name},
             "root": {
                 "dag": {
                     "tasks": {
-                        cleaned_component_name: {
+                        kfp_safe_name: {
                             "cachingOptions": {"enableCache": True},
-                            "componentRef": {"name": "comp-" + cleaned_component_name},
+                            "componentRef": {"name": "comp-" + kfp_safe_name},
                             "inputs": {
                                 "parameters": {
                                     param: {"componentInputParameter": param}
                                     for param in input_definitions["parameters"]
                                 },
                             },
-                            "taskInfo": {"name": cleaned_component_name},
+                            "taskInfo": {"name": kfp_safe_name},
                         },
                     },
                 },
                 "inputDefinitions": input_definitions,
             },
             "schemaVersion": "2.1.0",
-            "sdkVersion": "kfp-2.0.1",
+            "sdkVersion": "kfp-2.6.0",
```

Comment:
Any side effects by increasing the version?

Reply:
No, now it is in line with the KFP version we use now.